### PR TITLE
style: Somewhat miscelaneous cleanup.

### PR DESCRIPTION
**Pull request by emilio**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20117)

### CODE REVIEW

**Comment by emilio** [comment]
r? @nox or @SimonSapin or @bholley 

This is mostly moving code around, but there's some improvements, like the id accessor no longer cloning the attributes, and such.

File: components/style/context.rs [hunk 1]
```diff
@@ -383,8 +384,8 @@ impl fmt::Display for TraversalStatistics {
 impl TraversalStatistics {
     /// Computes the traversal time given the start time in seconds.
     pub fn finish<E, D>(&mut self, traversal: &D, parallel: bool, start: f64)
-        where E: TElement,
-              D: DomTraversal<E>,
+    where E: TElement,
```
Comment (nox) [inline components/style/context.rs]: AFAIK all trait bounds must be on their on line indented by 4 after the `where`, but I don't care very much in such a massive PR if that is the only nit I come with.

File: components/style/matching.rs [hunk 2]
```diff
@@ -313,7 +316,7 @@ trait PrivateMatchMethods: TElement {
         // in addition to the unvisited styles.
 
         let mut tasks = UpdateAnimationsTasks::empty();
-        if self.needs_animations_update(context, old_values.as_ref(), new_values) {
+        if self.needs_animations_update(context, old_values.as_ref().map(|s| &**s), new_values) {
```
Comment (nox) [inline components/style/matching.rs]: Do you really need the asterisks here? Or even the `&`? `s` is a `&_` and the method isn't generic, right?

**Review by nox — APPROVED** [review]

File: components/style/context.rs (hunk 1 above)
Comment (emilio) [reply components/style/context.rs]: Whoops, missed it, will fix.

File: components/style/matching.rs (hunk 2 above)
Comment (emilio) [reply components/style/matching.rs]: I'll check, I just grabbed from a few lines above.

Comment (emilio) [reply components/style/matching.rs]: Err, below

Comment (emilio) [reply components/style/matching.rs]: Happens to be needed, because it's in a `map`... shrug

**Comment by emilio** [comment]
@bors-servo retry

 * builder out of space

**Pull request merged**